## Importing MSFragger results (including timsTOF PASEF) into Skyline

This tutorial covers spectral library building in Skyline from validated MSFragger search results. In this example, we used Skyline-daily (v 19.1.9.350) and two files from the timsTOF PASEF dataset published by [Meier et al](https://www.mcponline.org/content/early/2018/11/01/mcp.TIR118.000900), which can be downloaded from [ProteomeXchange](http://proteomecentral.proteomexchange.org/cgi/GetDataset?ID=PXD010012) ('HeLa_200ng_100ms_raw.zip' on the FTP site).

This tutorial can also be used to import Thermo data into Skyline, '.mzML' files can be used in place of '.d' and '.mgf'/'\_calibrated.mgf'. Additionally, results from DIA ('interact-pep.xml' files and pseudo-DDA '.mzML' files generated by the 'DIA-Umpire' workflow in [FragPipe](https://fragpipe.nesvilab.org/) can also be imported into Skyline using this method.

#### Note 1: 
To allow MSFragger write 'calibrated.mgf', users need to check `Write calibrated MGF` in FragPipe (or set `write_calibrated_mgf = 1` in 'fragger.params' if using the command line).

#### Note 2:
Users need to change `deisotope` to 0 in MSFragger to use 'calibrated.mgf' in spectral library building.

#### Set up the analysis
Make sure the PeptideProphet output files ('interact-pep.xml'), raw spectral files ('.d'), and .mgf files ('.mgf' or '\_calibrated.mgf') in the same directory or in the same parent directory. You will also need the protein sequence database filtered at 1% FDR which is output from Philosopher ('protein.fas') that you used in the search.

#### Create a new DDA document

Launch Skyline and select **'Import DDA Peptide Search'** from the Start Page. Follow the prompts to choose a name and location for the new document to be saved.

#### Import peptide results
Once the new document is saved, **'Add Files'** to select the 'interact-pep.xml' files for the search results you want to import. 



To generate a spectral library that is filtered to 1% ion FDR and 1% protein FDR, we recommend the following:

1) Specifying the peptide probability threshold that corresponds to 1% ion FDR as the **'Cut-off score'**. (This can be found in the FragPipe log or .log file, a cut-off of 0.9024 would be used in this example:
_INFO[15:56:25] Converged to 1.00 % FDR with 57143 Ions       decoy=576 threshold=**0.9024** total=57719_)

2) Using 'protein.fas', which is generated by Philosopher filtering and contains sequences of proteins that passed 1% protein-level FDR file, in the 'Add FASTA file' step below.

Once you've set a cut-off score, press **'Next'**, prompting Skyline to build the spectral library from these peptide search results.

![](https://raw.githubusercontent.com/Nesvilab/MSFragger/master/images/Skyline_PASEF_4.png)
<br>

#### Add spectral files
When the spectral library is built, Skyline should automatically detect the spectral files. Press **'Next'**. If Skyline detects that the files in your experiment are named with a common prefix, you have the option to shorten the result names at this point for easier comparison later on.

![](https://raw.githubusercontent.com/Nesvilab/MSFragger/master/images/Skyline_PASEF_5.png)
<br>

#### Add detected modifications
Skyline should automatically detect variable modifications from your search results. Select the suggested modifications and press **'Next'**.

![](https://raw.githubusercontent.com/Nesvilab/MSFragger/master/images/Skyline_PASEF_6.png)
<br>

#### Configure scan settings
Precursor charge states greater than 2 can be added at this point. Press **'Next'**.

![](https://raw.githubusercontent.com/Nesvilab/MSFragger/master/images/Skyline_PASEF_7.png)
<br>

#### Add FASTA file
**'Browse'** for the sequence database, and select the number of missed cleavages that was used in the database search. Press **'Next'** to insert the 'protein.fas' file output by FragPipe/Philosopher, which contains the sequences of proteins filtered to 1% FDR.

![](https://raw.githubusercontent.com/Nesvilab/MSFragger/master/images/Skyline_PASEF_8.png)
<br>

#### Filter targets
When the FASTA file is finished importing, select the desired filters, and  press **'OK'** to start the import.

![](https://raw.githubusercontent.com/Nesvilab/MSFragger/master/images/Skyline_PASEF_9.png)
<br>

#### Importing the results
The import will take a few minutes per file.

![](https://raw.githubusercontent.com/Nesvilab/MSFragger/master/images/Skyline_PASEF_10.png)
<br>

#### View results
Once the import is finished, you are ready to perform your Skyline analyses, select a peptide from the list to view. Quantification can be compared between runs via _View > Peak Areas > Replicate Comparison_. 

![](https://raw.githubusercontent.com/Nesvilab/MSFragger/master/images/Skyline_PASEF_12.png)

To see the two-dimensional ion mobility vs. m/z plot, click on the apex of the chromatogram.

![](https://raw.githubusercontent.com/Nesvilab/MSFragger/master/images/Skyline_PASEF_13.png)

<br>
<br>
### [FragPipe homepage](https://fragpipe.nesvilab.org/)
